src/Microsoft.AspNetCore.SignalR.Core/HubEndPoint.cs

```diff
@@ -126,6 +132,22 @@ public class HubEndPoint<THub> where THub : Hub
             }
         }
 
+        private async Task SendCloseAsync(HubConnectionContext connection, Exception exception)
+        {
+            CloseMessage closeMessage = exception == null
+                ? CloseMessage.Empty
+                : new CloseMessage($"Connection closed with an error. {exception.GetType().Name}: {exception.Message}");
```

Comment:
We need to be careful sending exception details from the server to the client by default.

Reply:
https://github.com/aspnet/SignalR/issues/1614